libs/features/mas/src/hydrate.js

```diff
@@ -219,9 +219,78 @@ export function processPrices(fields, merchCard, mapping) {
     appendSlot('prices', fields, merchCard, mapping);
 }
 
+function transformLinkToButton(linkElement, merchCard, aemFragmentMapping) {
```

Comment:
I think you just need to run processCTAs against description instead of doing this.